### ISSUE TITLE
[bug] Running QHub within home directory will delete files in home directory

### ISSUE THREAD

**Issue by costrouc** [Quansight/qhub#767]
## Describe the bug

If a user runs the typical install process from their home directory.

```shell
conda create -n qhub -c conda-forge qhub==0.3.12
conda activate qhub

qhub init ...
qhub deploy ...
```

This will delete file within the home folder. This is due to the logic in https://github.com/Quansight/qhub/blob/744060aeaf19c194d39fd418c9834790c21f9f00/qhub/render/__init__.py#L139.

## Acceptance Criterial

 - remove all `os.remove` and `shutil.rmtree` logic in QHub this is too much of a risk for qhub.

**Comment by danlester** [comment]
This was discussed before we introduced the 'cleaning' logic. The problem without it is that we can't remove files from the cookie-cutter template, and over time can end up with e.g. tf files in some deployments that aren't expected and could thus break the deploy. Or removing a .github workflow file from cookie-cutter has no effect on the deployment.

A safer version of what we have now might be to target `terraform-state`, `.github.`, `infrastructure`, `images` folders instead of the top-level tree.

Or go the previously-discussed route of rendering to a temp location, at least by default for most deployments.

Or render within a sub-folder.

**Comment by tylerpotts** [comment]
Short term solution: 
- [ ] Whitelist directories that are allowed to get deleted
- [ ] Print out top level absolute paths that are getting deleted IE `[terraform-state, .github, infrastructure, images]`
- [ ] Don't allow rendering in home directory (compare current working directory to `$HOME` env variable)

Long term ideas:
- Terraform CDK
- Write less files
- Reduce dependency on cookiecutter

**Comment by costrouc** [comment]
Assigning @iameskild. Please comment on the issue to give updates.

**Comment by iameskild** [comment]
PR submitted.
(apologies for the delay, client work has consumed most of the past week)

**Comment by costrouc** [comment]
No worries at all @iameskild ! Thanks for the PR. I'll review it

**Comment by toonarmycaptain** [comment]
This is also apparently an issue with `qhub destroy` eg `qhub destroy -c your_dir/qhub-config.yaml` will delete many things that it should not, including miniconda binaries, the authorised keys file on your Digital Ocean droplet etc

**Comment by iameskild** [comment]
@toonarmycaptain thank you for sharing! Assigning myself to investigate this further. If you have any additional information related to this specific issue, please do share :)

**Comment by iameskild** [comment]
@toonarmycaptain I believe the issue you describe will be resolved with the #800. However, you can get around this issue by passing the `--disable-render` flag:
```
qhub destroy -c your_dir/qhub-config.yaml --disable-render
```

**Comment by toonarmycaptain** [comment]
@iameskild cheers :) 
Good practice to just run it from inside the correct directory anyway, which works fine for me.
